### PR TITLE
fix(core): 19724 customization, markdown, css processing updates

### PR DESCRIPTION
**Pull request by vkozio**
@coderabbitai pause

## Release Notes

- **New Features**
  - Introduced a `MarkdownContent` component for rendering Markdown text with custom link and media handling.
  - Added a `MarkdownLink` component for enhanced navigation link management.
  - Implemented a `MarkdownMedia` component for rendering images and YouTube videos in Markdown.
  - Added a new `PagesDocument` component for dynamic document rendering, improving content management for "About" and "Cookies" pages.

- **Improvements**
  - Updated article layout with a new grid system for better responsiveness and content presentation.
  - Enhanced routing for "About" and "Cookies" pages using the `PagesDocument` component for dynamic content loading.
  - Updated various CSS styles for improved visual consistency and responsiveness across the application.
  - Introduced a detailed styling framework for Markdown content, enhancing presentation and usability.
  - Updated dependency versions to ensure compatibility and improved performance.

- **Documentation**
  - Added a Customization Guide for configuring the markdown processor and `PagesDocument` component.
  - Updated documentation for the structured markdown processor, detailing its capabilities and usage.

### CODE REVIEW

**Comment by vkozio** [comment]
@coderabbitai resolve
@coderabbitai summary

File: src/components/Layout/Article/Article.tsx [hunk 1]
```diff
@@ -7,8 +7,8 @@ export function Article({
   id = '',
 }: PropsWithChildren<{ className?: string; id?: string }>) {
   return (
-    <div className={s.mainWrap} id={id}>
-      <article className={`${s.content} ${className}`}>{children}</article>
-    </div>
+    <article className={`${s.article} ${className}`} id={id}>
```
Comment (dqunbp) [inline src/components/Layout/Article/Article.tsx:10]: Can we use `clsx` here?

File: src/core/pages/MarkdownLink.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+      goTo(href);
+      e.preventDefault();
+    },
+    [href],
+  );
+
+  if (isExternalLink(href)) {
+    return (
+      <a title={title} href={href} target="_blank" rel="noreferrer" className="external">
```
Comment (dqunbp) [inline src/core/pages/MarkdownLink.tsx:54]: Why dont we use the class names from css modules here? e.g. `s.external`

File: src/core/pages/MarkdownMedia.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+ * YouTube with custom params:
+ * ![Video title](https://youtube.com/watch?v=xyz::800,450,1)
+ * ```
+ */
+export function MarkdownMedia({
+  title,
+  alt,
+  src,
+}: {
+  title: string;
```
Comment (dqunbp) [inline src/core/pages/MarkdownMedia.tsx:36]: The double expanded lists of properties and their types look difficult to read. Can we separately determine the props type above?

File: src/core/pages/PagesDocument.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  id,
+}: PagesDocumentProps) {
+  const memoizedDoc = useMemo(() => doc, [doc]);
+  const data = usePromise(fetchPagesDocument, [memoizedDoc]);
+
+  return (
+    <Wrapper id={`app-pages-docid-${id}`}>
+      {data.map((element, index) => {
+        const Renderer = PagesDocumentElementRenderers[element.type];
+        return <Renderer key={index} {...element} />;
```
Comment (dqunbp) [inline src/core/pages/PagesDocument.tsx:74]: Using an index as a react key is not a reliable practice.

File: src/core/pages/MarkdownContent.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import { MarkdownLink } from './MarkdownLink';
+import { MarkdownMedia } from './MarkdownMedia';
+import { structureMarkdownContent } from './structuredMarkdown';
+
+type MarkdownContentProps = {
+  content: string;
+};
+
+export function MarkdownContent({ content }: MarkdownContentProps) {
+  const markdownOptions = useMemo(
```
Comment (Natallia-Harshunova) [inline src/core/pages/MarkdownContent.tsx]: There is no need to store this options locally inside component, you can place it outside , imho

File: src/core/pages/PagesDocument.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+type PagesDocumentProps = {
+  doc: PagesDocumentElement[];
+  wrapperComponent?: React.ComponentType<{ id: string; children?: React.ReactNode }>;
+  id: string;
+};
+
+type PagesDocumentElementProps = {
+  data: string;
+  className?: string;
```
Comment (Natallia-Harshunova) [inline src/core/pages/PagesDocument.tsx]: Does the `className` property in this types is used and make effect anywhere?

File: src/core/pages/PagesDocument.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    </div>
+  );
+}
+
+export function PagesDocument({
+  doc,
+  wrapperComponent: Wrapper = Article,
+  id,
+}: PagesDocumentProps) {
+  const memoizedDoc = useMemo(() => doc, [doc]);
```
Comment (Natallia-Harshunova) [inline src/core/pages/PagesDocument.tsx]: I think that memoization here is redundant as doc is already a prop

File: src/core/pages/MarkdownMedia.tsx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+}) {
+  const { originalUrl, params } = parseMediaParams(src);
+
+  // Handle YouTube embeds
+  if (isYoutubeUrl(originalUrl)) {
+    return (
+      <iframe
+        src={getYoutubeEmbedUrl(originalUrl)}
+        width={params?.width || 560}
+        height={params?.height || 315}
```
Comment (Natallia-Harshunova) [inline src/core/pages/MarkdownMedia.tsx:48]: Youtube videos aren't adapted for mobile breakpoint. Maybe we can make something to handle this?

Comment (vkozio) [reply src/core/pages/MarkdownMedia.tsx:48]: We will in dedicated story about mobile layout

File: src/core/pages/MarkdownLink.tsx (hunk 2 above)
Comment (vkozio) [reply src/core/pages/MarkdownLink.tsx:54]: to be able to customize via global stylesheet

File: src/core/pages/PagesDocument.tsx (hunk 4 above)
Comment (vkozio) [reply src/core/pages/PagesDocument.tsx:74]: it ok here

**Comment by vkozio** [comment]
@coderabbitai resolve
@coderabbitai summary

**Pull request closed without merging**